src/App.js

```diff
@@ -1,19 +1,51 @@
 import React from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const calcTotalLikes = (chatData) => {
+    return chatData.reduce((total, entry) => {
+      if (entry.liked) {
+        total += 1
+      }
+      return total;
+    }, 0)
+  };
+
+  const totalLikeTally = calcTotalLikes(chatData);
+
+  const updateChatData = (updatedChatData) => {
+    const entries = chatData.map(entry => {
+      if (entry.id === updatedChatData.id) {
+        return updatedChatData
+      } else {
+        return entry
+      };
+    });
+    setChatData(entries);
+  };
```

Comment:
After checking `chatEntry`, I noticed that ya'll are passing up a brand new object and then adding that to the `entries` list. 

While this approach is consistent with what's shown in Learn, it has the downside that a presentation component needs to have enough information to reconstruct an object (including data not required for display) and know how to mutate the new object in the process.

I prefer the approach where only the object id is passed to the event handler. Sending up an id allows the child component to stay simpler, as it only needs the id and any data required for actual display, leaving all the messy stuff to be handled by code that has more knowledge of what the full data might look like.

Reply:
Nice work! We can also refactor the conditional into a ternary as well:

```javascript
  const updateChatdata = (id) => {
    const entries = chatData.map(entry => 
       entry.id === id? { ...entry, liked: !entry.liked } : entry
    );

    setMessageData(entries);
  };